From 65e0b15c35ea91fb40569aeacd46a56bfc9fe912 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Tue, 19 Feb 2019 18:48:29 +0200
Subject: [PATCH 11/25] ucm-imx8m-mini: add support for eeprom layout comands

Add support for EEPROM and EEPROM layout commands for the
UCM-iMX8M-Mini.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 configs/ucm-imx8m-mini_defconfig | 3 +++
 include/configs/ucm-imx8m-mini.h | 9 +++++++++
 2 files changed, 12 insertions(+)

diff --git a/configs/ucm-imx8m-mini_defconfig b/configs/ucm-imx8m-mini_defconfig
index 36cfe1b..b4a9975 100644
--- a/configs/ucm-imx8m-mini_defconfig
+++ b/configs/ucm-imx8m-mini_defconfig
@@ -19,6 +19,8 @@ CONFIG_FASTBOOT_BUF_ADDR=0x42800000
 CONFIG_FASTBOOT_BUF_SIZE=0x40000000
 CONFIG_FASTBOOT_FLASH=y
 CONFIG_FASTBOOT_FLASH_MMC_DEV=0
+CONFIG_CMD_EEPROM=y
+CONFIG_CMD_EEPROM_LAYOUT=y
 CONFIG_CMD_MEMINFO=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_GPIO=y
@@ -32,6 +34,7 @@ CONFIG_OF_CONTROL=y
 CONFIG_ENV_IS_IN_MMC=y
 CONFIG_DM_GPIO=y
 CONFIG_DM_I2C=y
+CONFIG_DM_I2C_COMPAT=y
 CONFIG_SYS_I2C_MXC=y
 CONFIG_LED=y
 CONFIG_LED_STATUS=y
diff --git a/include/configs/ucm-imx8m-mini.h b/include/configs/ucm-imx8m-mini.h
index 1605ab7..f464ed8 100644
--- a/include/configs/ucm-imx8m-mini.h
+++ b/include/configs/ucm-imx8m-mini.h
@@ -319,6 +319,15 @@
 #define CONFIG_SYS_I2C_MXC_I2C3		/* enable I2C bus 3 */
 #define CONFIG_SYS_I2C_SPEED		100000
 
+/* EEPROM */
+#define CONFIG_ENV_EEPROM_IS_ON_I2C
+#define CONFIG_SYS_I2C_EEPROM_BUS		1
+#define CONFIG_SYS_I2C_SLAVE			0x00
+#define CONFIG_SYS_I2C_EEPROM_ADDR_LEN		1
+#define CONFIG_SYS_EEPROM_SIZE			256
+#define CONFIG_SYS_EEPROM_PAGE_WRITE_BITS	4
+#define CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS	5
+
 /* USB configs */
 #ifndef CONFIG_SPL_BUILD
 #define CONFIG_CMD_USB
-- 
1.9.1

